runtime/src/main/java/org/corfudb/runtime/object/transactions/OptimisticTransactionalContext.java

```diff
@@ -322,69 +230,60 @@ public long commitTransaction() throws TransactionAbortedException {
                 address, affectedStreams);
 
         if (address == -1L) {
-            log.debug("Transaction aborted due to sequencer rejecting request");
+            log.trace("Commit[{}] Rejected by sequencer", this);
             abortTransaction();
             throw new TransactionAbortedException();
         }
 
         super.commitTransaction();
         commitAddress = address;
 
-        // Update all proxies, committing the new address.
         tryCommitAllProxies();
-
+        log.trace("Commit[{}] Written to {}", this, address);
         return address;
     }
 
-
     /** Try to commit the optimistic updates to each proxy. */
     protected void tryCommitAllProxies() {
         // First, get the committed entry
         // in order to get the backpointers
         // and the underlying SMREntries.
-        final ILogData committedEntry = this.builder.getRuntime()
+        ILogData committedEntry = this.builder.getRuntime()
                 .getAddressSpaceView().read(commitAddress);
 
-        if (committedEntry.getType() == DataType.EMPTY) {
-            return;
-        }
-
         updateAllProxies(x -> {
+            log.trace("Commit[{}] Committing {}", this,  x);
+            // Commit all the optimistic updates
+            x.getUnderlyingObject().optimisticCommitUnsafe();
```

Comment:
the name optimisticCommitUnsafe is legacy, I would change to something like 'resetOptimisticStream'